src/Eccube/Controller/Admin/Product/ProductController.php

```diff
@@ -537,26 +537,28 @@ public function edit(Request $request, $id = null, RouterInterface $router, Cach
 
                 // 画像の削除
                 $delete_images = $form->get('delete_images')->getData();
+                $fs = new Filesystem();
                 foreach ($delete_images as $delete_image) {
-                    $ProductImage = $this->productImageRepository
-                        ->findOneBy(['file_name' => $delete_image]);
+                    $ProductImage = $this->productImageRepository->findOneBy([
+                        'Product' => $Product,
+                        'file_name' => $delete_image,
+                    ]);
 
-                    // 追加してすぐに削除した画像は、Entityに追加されない
                     if ($ProductImage instanceof ProductImage) {
                         $Product->removeProductImage($ProductImage);
                         $this->entityManager->remove($ProductImage);
-                    }
-                    $this->entityManager->persist($Product);
-                    $this->entityManager->flush();
+                        $this->entityManager->flush();
 
-                    if (!$this->productImageRepository->findOneBy(['file_name' => $delete_image])) {
-                        // 削除
-                        $fs = new Filesystem();
-                        $fs->remove($this->eccubeConfig['eccube_save_image_dir'] . '/' . $delete_image);
+                        // 他に同じ画像を参照する商品がなければ画像ファイルを削除
+                        if (!$this->productImageRepository->findOneBy(['file_name' => $delete_image])) {
+                            $fs->remove($this->eccubeConfig['eccube_save_image_dir'] . '/' . $delete_image);
+                        }
+
+                    } else {
+                        // 追加してすぐに削除した画像は、Entityに追加されない
+                        $fs->remove($this->eccubeConfig['eccube_temp_image_dir'] . '/' . $delete_image);
                     }
                 }
-                $this->entityManager->persist($Product);
-                $this->entityManager->flush();
 
```

Comment:
@izayoi256 

```suggestion
                $this->entityManager->flush();
```

新しい画像を登録する際、一度ここで永続化しておかないと、567行目の `findOneBy` で Entity が取得できずにエラーとなりました。